lib/client/player.go

```diff
@@ -31,8 +31,17 @@ import (
 type tshPlayerState int
 
 const (
+	// The player has stopped for an action to take place,
+	// or the playback has ended either by request or because
+	// it was played all the way through. setState(stateStopped)
+	// should only be called from playRange once it's goroutine has
+	// returned.
 	stateStopped tshPlayerState = iota
+	// A stop has been requested so that an action can take place.
 	stateStopping
+	// An end to the playback has been requested.
+	stateEnding
+	// The player is playing.
 	statePlaying
 )
 
```

Comment:
I can't comment in the right spot, but let's add a comment where `stopC` is declared that says:

```
// stopC is called when playback ends (either because the end of the stream has been reached, or a hard stop was requested via Stop())
```